File: Java/5. Longest Palindromic Substring.java
class Solution {
    public String longestPalindrome(String s) {
        int n=s.length();
        int len=1;
        int begin=0;
        for(int i=1;i<=n;i++){
            int l=i-1;
            int r=i;
            while(l>=0 && r<n && s.charAt(l)==s.charAt(r)){
                if(r-l+1>len){
                    len=r-l+1;
                    begin=l;
                }
                l--;
                r++;
            }
            l=i-1;
            r=i+1;
            while(l>=0 && r<n && s.charAt(l)==s.charAt(r)){
                if(r-l+1>len){
                    len=r-l+1;
                    begin=l;
                }
                l--;
                r++;
            }
        }
        return s.substring(begin,begin+len);
    }
}

File: JavaScript/5. Longest Palindromic Substring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    var n=s.length;
    var len=1;
    var begin=0;
    for(var i=1;i<=n;i++){
        var l=i-1;
        var r=i;
        while(l>=0 && r<n && s[l]==s[r]){
            if(r-l+1>len){
                len=r-l+1;
                begin=l;
            }
            l--;
            r++;
        }
        l=i-1;
        r=i+1;
        while(l>=0 && r<n && s[l]==s[r]){
            if(r-l+1>len){
                len=r-l+1;
                begin=l;
            }
            l--;
            r++;
        }
    }
    return s.substring(begin,begin+len);
};
